test/core/compression/compression_test.c

```diff
@@ -71,9 +70,93 @@ static void test_compression_algorithm_parse(void) {
   }
 }
 
+static void test_compression_algorithm_name(void) {
+  int success;
+  char *name;
+  size_t i;
+  const char *valid_names[] = {"identity", "gzip", "deflate"};
+  const grpc_compression_algorithm valid_algorithms[] = {
+      GRPC_COMPRESS_NONE, GRPC_COMPRESS_GZIP, GRPC_COMPRESS_DEFLATE};
+
+  gpr_log(GPR_DEBUG, "test_compression_algorithm_name");
+
+  for (i = 0; i < GPR_ARRAY_SIZE(valid_algorithms); i++) {
+    success = grpc_compression_algorithm_name(valid_algorithms[i], &name);
+    GPR_ASSERT(success != 0);
+    GPR_ASSERT(strcmp(name, valid_names[i]) == 0);
+  }
+
+  success =
+      grpc_compression_algorithm_name(GRPC_COMPRESS_ALGORITHMS_COUNT, &name);
+  GPR_ASSERT(success == 0);
+  /* the value of "name" is undefined upon failure */
+}
+
+
+static void test_compression_algorithm_for_level(void) {
+  size_t i;
+  grpc_compression_level levels[] = {
+      GRPC_COMPRESS_LEVEL_NONE, GRPC_COMPRESS_LEVEL_LOW,
+      GRPC_COMPRESS_LEVEL_MED, GRPC_COMPRESS_LEVEL_HIGH};
+  grpc_compression_algorithm algorithms[] = {GRPC_COMPRESS_NONE,
+    GRPC_COMPRESS_DEFLATE, GRPC_COMPRESS_DEFLATE, GRPC_COMPRESS_DEFLATE};
+  gpr_log(GPR_DEBUG, "test_compression_algorithm_for_level");
+
+  for (i = 0; i < GPR_ARRAY_SIZE(levels); i++) {
+    GPR_ASSERT(algorithms[i] ==
+               grpc_compression_algorithm_for_level(levels[i]));
+  }
+}
+
+static void test_compression_level_for_algorithm(void) {
```

Comment:
Lets kill this function (grpc_compression_level_for_algorithm) - it's impossible to use correctly